dev/PushNotifications/PushNotificationReceivedEventArgs.cpp

```diff
@@ -0,0 +1,113 @@
+﻿#include "pch.h"
+
+#include <winrt/Windows.ApplicationModel.background.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <winrt/Windows.Networking.PushNotifications.h>
+
+#include "PushNotificationReceivedEventArgs.h"
+#include "Microsoft.Windows.PushNotifications.PushNotificationReceivedEventArgs.g.cpp"
+#include <iostream>
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Networking::PushNotifications;
+
+namespace winrt::Microsoft::Windows::PushNotifications::implementation
+{
+    PushNotificationReceivedEventArgs::PushNotificationReceivedEventArgs(winrt::Windows::ApplicationModel::Background::IBackgroundTaskInstance const& backgroundTask)
+    {
+        m_backgroundTaskInstance = backgroundTask;
+
+        auto triggerDetails = m_backgroundTaskInstance.TriggerDetails();
+        RawNotification rawNotification = triggerDetails.as<RawNotification>();
+
+        IBuffer rawContentAsBuffer = rawNotification.ContentBytes();
+        DataReader dataReader = DataReader::FromBuffer(rawContentAsBuffer);
+
+        m_backgroundPayload = winrt::com_array<uint8_t>(rawContentAsBuffer.Length());
```

Comment:
Can you confirm why we need m_payload if we already have m_backgroundTaskInstance?

Reply:
I messaged the owner of BackgroundTaskInstance and he said that by design, when BackgroundTaskInstance exits the run function, the TriggerDetails property will be deallocated. He said that if we need to use the payload later, to store it before it exits the Run function.

Reply:
Please add a comment then right above this explaining the same